### PR TITLE
Commerce 2468

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test

File: headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/WarehouseItemResourceImpl.java [hunk 1]
```diff
@@ -167,8 +167,7 @@ public WarehouseItem getWarehouseItemByExternalReferenceCode(
 		}
 
 		if ((start == null) && (end == null)) {
-			throw new CommerceInventoryInvalidDateException(
-				"At least one date should be defined");
+			start = new Date();
```
Comment (andrea-ale-sbarra) [inline headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/WarehouseItemResourceImpl.java:172]: @marco-leo not 100% about this change, but I think it make sense.. let me know what do you think. Thank you.

File: headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/WarehouseItemResourceImpl.java [hunk 2]
```diff
@@ -232,51 +304,6 @@ public WarehouseItem postWarehousByExternalReferenceCodeWarehouseItem(
 					getCommerceInventoryWarehouseItemId()));
 	}
 
-	@Override
-	public WarehouseItem postWarehouseItemByExternalReferenceCode(
```
Comment (andrea-ale-sbarra) [inline headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/WarehouseItemResourceImpl.java]: @marco-leo also about this I'm not 100% sure, but in the end I think the use case is well covered with other methods, plus this was very tricky and confusing.. Let me know what do you think about this. Thank you!

**Comment by andrea-ale-sbarra** [comment]
@ethanbustad Sorry ethan, but I don't really understand these issues, please help me

**Comment by marco-leo** [comment]
@andrea-ale-sbarra let's talk over those changes

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
@marco-leo I restored the post method and fixed the validation on that

File: headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/WarehouseItemResourceImpl.java [hunk 3]
```diff
@@ -65,6 +67,31 @@ public Response deleteWarehouseItem(Long id) throws Exception {
...
+			throw new NoSuchInventoryWarehouseItemException(
+				"Unable to find WarehouseItem with externalReferenceCode: " +
+					externalReferenceCode);
+		}
+
+		_commerceInventoryWarehouseItemService.
+			deleteCommerceInventoryWarehouseItem(
+				commerceInventoryWarehouseItem.
+					getCommerceInventoryWarehouseItemId());
+		Response.ResponseBuilder responseBuilder = Response.noContent();
```
Comment (marco-leo) [inline headless/headless-commerce/headless-commerce-admin-inventory-impl/src/main/java/com/liferay/headless/commerce/admin/inventory/internal/resource/v1_0/WarehouseItemResourceImpl.java:91]: Empty line

**Review by marco-leo — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by ethanbustad** [comment]
Hey @andrea-ale-sbarra, no problem, these are unusual errors :)

Any time you see something like `Could not GET 'http://repository-cdn.liferay.com`, it's just a network error in CI -- just execute the tests again.

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Pull request merged**